src/libmongoc/tests/test-mongoc-bulk.c

```diff
@@ -476,30 +476,13 @@ test_insert_check_keys (void)
    BSON_ASSERT (bulk);
 
    mongoc_bulk_operation_insert (bulk, tmp_bson (NULL));
-   mongoc_bulk_operation_insert (bulk, tmp_bson ("{'$dollar': 1}"));
+   mongoc_bulk_operation_insert (bulk, tmp_bson ("{'': 1}"));
    r = (bool) mongoc_bulk_operation_execute (bulk, &reply, &error);
    BSON_ASSERT (!r);
    ASSERT_ERROR_CONTAINS (error,
                           MONGOC_ERROR_COMMAND,
                           MONGOC_ERROR_COMMAND_INVALID_ARG,
-                          "keys cannot begin with \"$\": \"$dollar\"");
-
-   BSON_ASSERT (bson_empty (&reply));
-
-   bson_destroy (&reply);
-   mongoc_bulk_operation_destroy (bulk);
-
-   /* keys can't contain "." */
-   bulk = mongoc_collection_create_bulk_operation_with_opts (collection, NULL);
-   BSON_ASSERT (bulk);
-
-   mongoc_bulk_operation_insert (bulk, tmp_bson ("{'a.b': 1}"));
-   r = (bool) mongoc_bulk_operation_execute (bulk, &reply, &error);
-   BSON_ASSERT (!r);
-   ASSERT_ERROR_CONTAINS (error,
-                          MONGOC_ERROR_COMMAND,
-                          MONGOC_ERROR_COMMAND_INVALID_ARG,
-                          "keys cannot contain \".\": \"a.b\"");
```

Comment:
There was nothing to replace this test with, since we already use empty keys earlier and invalid UTF-8 isn't an option (due to how `tmp_bson` parses JSON).